cpp/tests/utilities/base_fixture.hpp

```diff
@@ -234,12 +237,14 @@ class TempDirTestEnvironment : public ::testing::Environment {
  *        Accepted types are "pool", "cuda", and "managed" only.
  * @return Memory resource instance
  */
-inline std::unique_ptr<rmm::mr::device_memory_resource> create_memory_resource(
-  std::string const &allocation_mode)
+inline rmm::mr::device_memory_resource *create_memory_resource(std::string const &allocation_mode)
 {
-  if (allocation_mode == "cuda") return std::make_unique<rmm::mr::cuda_memory_resource>();
-  if (allocation_mode == "pool") return std::make_unique<rmm::mr::cnmem_memory_resource>();
-  if (allocation_mode == "managed") return std::make_unique<rmm::mr::managed_memory_resource>();
+  if (allocation_mode == "cuda") return new rmm::mr::cuda_memory_resource();
+  if (allocation_mode == "pool") {
+    auto cuda_mr = new rmm::mr::cuda_memory_resource;
```

Comment:
Maybe make `cuda_mr` function static here? Since it is essentially global to the test process.

Reply:
Maybe. Waiting on Jake's new owning_wrapper.